Percy/providers/GenericProvider.cs

```diff
@@ -76,20 +79,162 @@ internal void SetDebugUrl(string debugUrl)
+          var ignoredRegion = IgnoreElementObject(selector, location, size);
+          ignoredElementsArray.Add(ignoredRegion);
+        } catch(Exception e) {
+          AppPercy.Log("Appium Element with xpath:" + xpath + " not found. Ignoring this xpath.");
+          AppPercy.Log(e.ToString());
+        }
+      }
+    }
+
+    public void IgnoreRegionsByIds(JArray ignoredElementsArray, List<String> ids)
+    {
+      foreach (var id in ids)
+      {
+        try
+        {
+          var element = percyAppiumDriver.FindElementsByAccessibilityId(id);
+
+          var location = element.Location;
+          var size = element.Size;
+          var selector = string.Format("id: {0}", id);
+          var ignoredRegion = IgnoreElementObject(selector, location, size);
+          ignoredElementsArray.Add(ignoredRegion);
+        } catch (Exception e) {
+          AppPercy.Log("Appium Element with id:" + id + " not found. Ignoring this id.");
+          AppPercy.Log(e.ToString());
+        }
+      }
+    }
+
+    public void IgnoreRegionsByElement(JArray ignoredElementsArray, List<AppiumWebElement> elements)
+    {
+      for (var index = 0; index < elements.Count; index++)
+      {
+        try
+        {
+          var location = elements[index].Location;
+          var size = elements[index].Size;
+          string type = elements[index].GetAttribute("class");
+          var selector = string.Format("element: {0} {1}", index, type);
+            
+          var ignoredRegion = IgnoreElementObject(selector, location, size);
+          ignoredElementsArray.Add(ignoredRegion);
+        } catch (Exception e) {
+          AppPercy.Log("Correct Appium Element not passed at index " + index + ".");
+          AppPercy.Log(e.ToString(), "debug");
+        }
+      }
+    }
+
+    public void AddCustomIgnoreRegion(JArray ignoredElementsArray, List<IgnoreRegion> customLocations)
+    {
+      var width = metadata.DeviceScreenWidth();
+      var height = metadata.DeviceScreenHeight();
+      for (var index = 0; index < customLocations.Count; index++)
+      {
+        try
+        {
+          if (ValidateIgnoreLocation(customLocations[index])) {
+            var selector = "custom ignore region " + index;
+            var ignoredRegion = JObject.FromObject(new
+            {
+              selector = selector,
+              co_ordinates = JObject.FromObject(
+                customLocations[index]
+              )
+            });
+            ignoredElementsArray.Add(ignoredRegion);
+          }
+          else
+            AppPercy.Log("Values passed in custom ignored region at index:- "+ index+ " is not valid");
+        } catch (Exception e)
+        {
+          AppPercy.Log("Custom Ignore Region object not valid at index:- "+ index);
+          AppPercy.Log(e.ToString(), "debug");
+        }
+        index++;
+      }
+    }
+
+    public Boolean ValidateIgnoreLocation(IgnoreRegion customLocation)
```

Comment:
```suggestion
    public Boolean ValidateIgnoreRegion(IgnoreRegion customLocation)
```
Also its better to just add this validation on IgnoreRegion class itself, so that you can just do `ignoreRegion.IsValid()`

Reply:
done create IsValid function inside IgnoreRegion class